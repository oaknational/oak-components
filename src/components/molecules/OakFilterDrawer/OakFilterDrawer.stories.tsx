import React from "react";
import { StoryObj, Meta } from "@storybook/react";
import { useArgs } from "@storybook/preview-api";

import { OakFilterDrawer } from "./OakFilterDrawer";

import { OakSecondaryButton } from "@/components/molecules/OakSecondaryButton";
import { OakBox, OakFlex, OakHeading } from "@/components/atoms";
import { OakSearchFilterCheckBox } from "@/components/organisms";
import { OakRadioGroup } from "@/components/molecules/OakRadioGroup";
import { OakPrimaryButton } from "@/components/molecules/OakPrimaryButton";
import { OakRadioButton } from "@/components/molecules/OakRadioButton";

const meta: Meta<typeof OakFilterDrawer> = {
  component: OakFilterDrawer,
  tags: ["autodocs"],
  title: "components/molecules/OakFilterDrawer",
  argTypes: {
    children: {
      control: "text",
    },
  },
  parameters: {
    controls: {
      include: ["children", "isOpen", "onClose", "clearAllInputs"],
    },
  },
  args: {
    "aria-label": "Example filter drawer",
    "aria-description": "This is an example filter drawer",
    children: (
      <>
        <OakHeading $mv={"space-between-m"} tag={"h4"}>
          Filter option buttons
        </OakHeading>
        <OakFlex
          $flexWrap={"wrap"}
          $mb={"space-between-m2"}
          $gap={"space-between-m"}
        >
          {[1, 2, 3].map((i) => {
            return (
              <OakSearchFilterCheckBox
                key={i}
                onChange={() => {}}
                value={`filter-option-${i}`}
                id={`filter-option-${i}`}
                displayValue={`${i} filter option`}
              />
            );
          })}
        </OakFlex>
        <OakHeading $mb={"space-between-m"} tag="h4">
          Filter option radio
        </OakHeading>
        <OakBox $mb={"space-between-m"}>
          <OakRadioGroup
            $gap={"space-between-m"}
            name="radio-group"
            $flexWrap={"wrap"}
          >
            <OakRadioButton id="radio-1" value="1" label="Option 1" />
            <OakRadioButton id="radio-2" value="2" label="Option 2" />
            <OakRadioButton id="radio-3" value="3" label="Option 3" />
            <OakRadioButton id="radio-4" value="4" label="Option 4" />
            <OakRadioButton id="radio-5" value="5" label="Option 5" />
          </OakRadioGroup>
        </OakBox>
      </>
    ),
  },
  render: (args) => {
    const [, updateArgs] = useArgs();
    const onClose = () => updateArgs({ isOpen: false });
    const onOpen = () => updateArgs({ isOpen: true });

    return (
      <>
        <OakSecondaryButton onClick={onOpen}>
          Open filter drawer
        </OakSecondaryButton>
        <OakFilterDrawer
          {...args}
          onClose={onClose}
          footerSlot={
            <OakPrimaryButton onClick={onClose}>
              Show results (23)
            </OakPrimaryButton>
          }
        />
      </>
    );
  },
};

export default meta;

type Story = StoryObj<typeof OakFilterDrawer>;

export const Default: Story = {};
