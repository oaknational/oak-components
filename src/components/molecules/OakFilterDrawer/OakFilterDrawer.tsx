import React, { useRef, useState } from "react";
import { createPortal } from "react-dom";
import styled from "styled-components";

import { InternalShadowRoundButton } from "@/components/molecules/InternalShadowRoundButton";
import { OakBox, OakFlex, OakHeading } from "@/components/atoms";
import { OakModalProps, OakSecondaryLink } from "@/components/molecules";
import useCanaryObserver from "@/hooks/useCanaryObserver";
import useMounted from "@/hooks/useMounted";
import InternalModalTransition from "@/components/molecules/InternalModalTransition/InternalModalTransition";

type OakFilterDrawerProps = OakModalProps & {
  /**
   * Called when the clear button is clicked
   */
  clearAllInputs: () => void;
};

const StyledClearButton = styled(OakSecondaryLink)`
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`;

/**
 *
 * Based on the OakModal component, this component is a filter drawer that slides in from the right side of the screen.
 * Designed for mobile and tablet devices
 *
 * ### Callbacks
 * clearAllInputs: used to clear filter options
 *
 */

export const OakFilterDrawer = ({
  isOpen,
  domContainer,
  onClose,
  zIndex,
  children,
  clearAllInputs,
  footerSlot,
  ...rest
}: OakFilterDrawerProps) => {
  const [canaryElement, setCanaryElement] = useState<HTMLDivElement | null>(
    null,
  );
  const isScrolled = useCanaryObserver(canaryElement);

  const transitionRef = useRef<HTMLDivElement>(null);

  const isMounted = useMounted();

  if (!isMounted) {
    return null;
  }
  const finalZIndex = typeof zIndex === "number" ? zIndex : "modal-dialog";

  return createPortal(
    <InternalModalTransition
      isOpen={isOpen}
      transitionRef={transitionRef}
      onClose={onClose}
      finalZIndex={finalZIndex}
      isModal={false}
      {...rest}
    >
      <OakFlex
        $ma="space-between-m"
        $justifyContent="space-between"
        $alignItems="center"
      >
        <StyledClearButton
          onClick={clearAllInputs}
          aria-label="Clear"
          element="button"
        >
          Clear
        </StyledClearButton>
        <OakHeading $font={"heading-6"} tag="h3">
          Filters
        </OakHeading>
        <InternalShadowRoundButton
          onClick={onClose}
          aria-label="Close"
          defaultIconBackground="transparent"
          defaultIconColor="black"
          defaultTextColor="transparent"
          hoverTextColor="transparent"
          disabledTextColor="transparent"
          hoverIconBackground="black"
          hoverIconColor="white"
          disabledIconBackground="transparent"
          iconBackgroundSize="all-spacing-6"
          iconSize="all-spacing-6"
          iconName="cross"
        />
      </OakFlex>
      <OakFlex
        $flexGrow={1}
        $flexDirection="column"
        $overflow="auto"
        $bt="border-solid-s"
        $borderColor={isScrolled ? "border-neutral-lighter" : "transparent"}
      >
        <div ref={setCanaryElement} />
        <OakBox $mh="space-between-m">{children}</OakBox>
      </OakFlex>
      <OakFlex
        $flexDirection={["column", "row"]}
        $bt="border-solid-s"
        $borderColor="border-neutral-lighter"
        $pa="inner-padding-s"
        $gap={["space-between-s", "space-between-m"]}
        $width="100%"
      >
        {footerSlot}
      </OakFlex>
    </InternalModalTransition>,
    domContainer ?? document.body,
  );
};
