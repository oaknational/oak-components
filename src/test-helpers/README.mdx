# test-helpers

Includes conveniences to ease in testing with Oak Components.

### `installMockResizeObserver`

Installs a mock [`ResizeObserver`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) that can be used to test components that use `ResizeObserver` in an environment where it isn't present.

### `installMockIntersectionObserver`

Installs a mock [`IntersectionObserver`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver) that can be used to test components that use `IntersectionObserver` in an environment where it isn't present.
