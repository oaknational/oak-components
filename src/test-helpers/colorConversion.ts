export const rgbToHex = (rgb: string) => {
  const match = rgb.match(/\d+/g); // Extract numbers from RGB format
  if (!match || match.length < 3) return rgb; // Ensure we have at least R, G, B values

  return `#${match
    .slice(0, 3) // Use only first 3 values (ignore alpha if present)
    .map((x) => {
      const hex = parseInt(x, 10).toString(16); // Convert to hex
      return hex.length === 1 ? "0" + hex : hex; // Ensure two-digit hex values
    })
    .join("")}`;
};
