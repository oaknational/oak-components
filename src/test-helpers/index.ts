export * from "./mockIntersectionObserver";
export * from "./mockResizeObserver";
