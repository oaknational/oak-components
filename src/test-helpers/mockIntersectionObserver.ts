/**
 * Installs a mock of IntersectionObserver when it is not present in the environment
 *
 * This is a global mock, so it will affect all tests that run after it is imported.
 */
export function installMockIntersectionObserver() {
  global.IntersectionObserver =
    global.IntersectionObserver ??
    class MockIntersectionObserver implements IntersectionObserver {
      private elements: Set<Element> = new Set();

      get root(): Element {
        throw new Error("Attribute not implemented.");
      }
      get rootMargin(): string {
        throw new Error("Attribute not implemented.");
      }
      get thresholds(): number[] {
        throw new Error("Attribute not implemented.");
      }
      takeRecords(): IntersectionObserverEntry[] {
        throw new Error("Method not implemented.");
      }
      observe(element: Element) {
        this.elements.add(element);
        // console.log(`Observing element: ${this.describeElement(element)}`);
      }
      unobserve(element: Element) {
        this.elements.delete(element);
        // console.log(`Stopped observing: ${this.describeElement(element)}}`);
      }
      disconnect() {
        this.elements.clear();
        // console.log("Disconnected all observed elements.");
      }
      // private describeElement(element: Element): string {
      //   let description = element.tagName.toLowerCase();
      //   if (element.id) {
      //     description += "#" + element.id;
      //   }
      //   if (element.className) {
      //     description += "." + element.className.replace(/\s+/g, ".");
      //   }
      //   return description;
      // }
    };
}
