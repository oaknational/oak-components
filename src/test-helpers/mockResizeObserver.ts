/**
 * Installs a mock of ResizeObserver when it is not present in the environment
 *
 * This is a global mock, so it will affect all tests that run after it is imported.
 */

export function installMockResizeObserver() {
  global.ResizeObserver =
    global.ResizeObserver ??
    class MockResizeObserver implements ResizeObserver {
      disconnect() {}
      observe() {}
      unobserve() {}
    };
}
