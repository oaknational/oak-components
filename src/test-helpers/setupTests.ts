import "jest-styled-components";

process.env.NEXT_PUBLIC_OAK_ASSETS_HOST = "res.cloudinary.com";
process.env.NEXT_PUBLIC_OAK_ASSETS_PATH = "mock-cloudinary-cloud/image/upload";
process.env.NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME = "mock-cloudinary-cloud";

// Override this with `TEST_ALLOW_LOGGING=1` if you want logs locally
if (process.env.TEST_ALLOW_LOGGING !== "1") {
  const consoleMethods = ["log", "info", "error", "warn", "debug"] as const;
  type ConsoleMethod = (typeof consoleMethods)[number];

  consoleMethods.forEach((type: ConsoleMethod) => {
    console[type] = (message: string) => {
      throw new Error(
        `Failed: We don't allow console.${type} while running tests!\n\n${message}`,
      );
    };
  });
}
