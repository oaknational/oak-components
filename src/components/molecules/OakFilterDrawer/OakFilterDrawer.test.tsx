import React, { ReactNode } from "react";
import "@testing-library/jest-dom";
import { act, create } from "react-test-renderer";
import { fireEvent } from "@testing-library/react";

import { OakFilterDrawer } from "./OakFilterDrawer";

import renderWithTheme from "@/test-helpers/renderWithTheme";
import { installMockIntersectionObserver } from "@/test-helpers";

installMockIntersectionObserver();

jest.mock("react-dom", () => {
  return {
    ...jest.requireActual("react-dom"),
    createPortal: (node: ReactNode) => node,
  };
});

describe("OakFilterDrawer", () => {
  it("renders component", () => {
    const { getByTestId } = renderWithTheme(
      <OakFilterDrawer
        data-testid="test"
        onClose={() => {}}
        clearAllInputs={() => {}}
        isOpen={true}
        children="Click Me"
      />,
    );
    expect(getByTestId("test")).toBeInTheDocument();
  });

  it("calls onClose when the close button is clicked", () => {
    const onCloseSpy = jest.fn();

    const { getByLabelText } = renderWithTheme(
      <OakFilterDrawer isOpen clearAllInputs={() => {}} onClose={onCloseSpy}>
        Filter content
      </OakFilterDrawer>,
    );

    act(() => {
      fireEvent.click(getByLabelText("Close"));
    });

    expect(onCloseSpy).toHaveBeenCalled();
  });

  it("gives the first focusable element in the modal body focus", () => {
    const { getByRole } = renderWithTheme(
      <OakFilterDrawer isOpen clearAllInputs={() => {}} onClose={() => {}}>
        <input type="text" />
      </OakFilterDrawer>,
    );

    expect(getByRole("textbox")).toHaveFocus();
  });

  it("matches snapshot", () => {
    const tree = create(
      <OakFilterDrawer
        clearAllInputs={() => {}}
        onClose={() => {}}
        isOpen={true}
      >
        Click Me
      </OakFilterDrawer>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
