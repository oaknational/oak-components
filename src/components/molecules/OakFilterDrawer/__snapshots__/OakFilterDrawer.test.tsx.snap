// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakFilterDrawer matches snapshot when mounted 1`] = `
.c0 {
  font-family: --var(google-font),Lexend,sans-serif;
  z-index: 300;
}

.c0:hover {
  cursor: pointer;
}

.c2 {
  position: fixed;
  top: 0rem;
  bottom: 0rem;
  left: 0rem;
  width: 40rem;
  color: #222222;
  background: #ffffff;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  font-family: --var(google-font),Lexend,sans-serif;
  z-index: 300;
}

.c5 {
  margin: 1.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c11 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c16 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c18 {
  position: relative;
  width: 1.5rem;
  min-width: 1.5rem;
  height: 1.5rem;
  color: transparent;
  background: transparent;
  border-radius: 6.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c20 {
  position: absolute;
  top: 0rem;
  width: 100%;
  height: 100%;
  border-radius: 6.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c21 {
  position: relative;
  width: 1.5rem;
  min-width: 1.5rem;
  height: 1.5rem;
  min-height: 1.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c24 {
  overflow: auto;
  border-top: 0.063rem solid;
  border-color: transparent;
  font-family: --var(google-font),Lexend,sans-serif;
  overflow: auto;
}

.c26 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c27 {
  width: 100%;
  padding: 0.75rem;
  border-top: 0.063rem solid;
  border-color: #cacaca;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c22 {
  -webkit-filter: invert(10%) sepia(1%) saturate(236%) hue-rotate(314deg) brightness(95%) contrast(91%);
  filter: invert(10%) sepia(1%) saturate(236%) hue-rotate(314deg) brightness(95%) contrast(91%);
  object-fit: contain;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0rem;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c28 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.c10 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c9 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c23 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c14 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  text-align: left;
  font-family: unset;
  outline: none;
  font-family: --var(google-font),Lexend,sans-serif;
  color: transparent;
}

.c14:disabled {
  pointer-events: none;
  cursor: default;
}

.c15 {
  display: inline-block;
  position: relative;
}

.c15:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: transparent;
}

.c15:hover:not(:active) [data-icon-for="button"] img {
  -webkit-filter: invert(98%) sepia(98%) saturate(0%) hue-rotate(328deg) brightness(102%) contrast(102%);
  filter: invert(98%) sepia(98%) saturate(0%) hue-rotate(328deg) brightness(102%) contrast(102%);
}

.c15:active {
  color: transparent;
}

.c15:disabled {
  color: transparent;
}

.c13 > :first-child:focus-visible .shadow {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c13 > :first-child:hover .shadow {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%);
}

.c13 > :first-child:active .shadow {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%), 0.25rem 0.25rem 0 rgba(87,87,87,100%);
}

.c13 > :first-child:disabled .icon-container {
  background: transparent;
}

.c13 > :first-child:hover .icon-container {
  background: #222222;
}

.c13 > :first-child:active .icon-container {
  background: transparent;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  outline: none;
  border-radius: 0.375rem;
  padding: 0.25rem;
  margin: -0.25rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  background: none;
  border: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  color: #222222;
}

.c7:focus-visible {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c7:visited {
  color: #575757;
}

.c7:active {
  color: #222222;
}

.c7[disabled] {
  cursor: not-allowed;
  color: #808080;
}

.c4 {
  max-width: 100vw;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c1 {
  opacity: 0.25;
  background-color: black;
  position: fixed;
  inset: 0;
  -webkit-transition: ease;
  transition: ease;
}

.c8 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media (min-width:750px) {
  .c28 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:750px) {
  .c28 {
    gap: 1.5rem;
  }
}

@media (hover:hover) {
  .c7:hover,
  .c7:visited:hover {
    color: #222222;
  }
}

@media (min-width:768px) {
  .c4 {
    max-width: 600px;
  }
}

<div>
  <div
    aria-hidden="true"
    data-focus-guard="true"
    data-focus-on-hidden="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="c0 c1"
    />
    <div
      class="c2 c3 c4"
      role="dialog"
    >
      <div
        class="c5 c6"
      >
        <button
          aria-label="Clear"
          class="c7 c8"
        >
          <span
            class="c9"
          >
            Clear
          </span>
        </button>
        <h3
          class="c10"
        >
          Filters
        </h3>
        <div
          class="c11 c12 c13"
        >
          <button
            aria-label="Close"
            class="c14 c15"
          >
            <div
              class="c16 c17"
            >
              <div
                class="c18 c19 icon-container"
              >
                <div
                  class="c20 shadow"
                />
                <div
                  class="c21"
                  data-icon-for="button"
                >
                  <img
                    alt="cross"
                    class="c22"
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699895179/icons/xigimrbivcaxt4omxamp.svg"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                  />
                </div>
              </div>
              <span
                class="c23"
              />
            </div>
          </button>
        </div>
      </div>
      <div
        class="c24 c25"
      >
        <div />
        <div
          data-autofocus-inside="true"
          tabindex="-2"
        >
          <div
            class="c26"
          >
            Modal content
          </div>
        </div>
      </div>
      <div
        class="c27 c28"
      >
        Oak filter footer
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    data-focus-guard="true"
    data-focus-on-hidden="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;
